### PR TITLE
Add `NO_EXPORT` option

### LINKED ISSUE #263: `install_basic_package_files()`: add `NO_EXPORT` option
`install_basic_package_files()` currently requires an export to be specified. However, sometimes we want to install but don't have an export set or any targets to export, e.g. in a CMake script library such as YCM itself.

Would you consider adding a `NO_EXPORT` option to `install_basic_package_files()` which makes it skip the `install()` and `export()` commands related to the export set? I'd be happy to provide a pull request.

### PR DESCRIPTION
**Pull request by mbeutel**
Closes #263

### CODE REVIEW

File: modules/InstallBasicPackageFiles.cmake [hunk 1]
```diff
@@ -386,6 +388,12 @@ function(INSTALL_BASIC_PACKAGE_FILES _Name)
     set(_IBPF_COMPONENT "${_Name}")
   endif()
 
+  if(_IBPF_NO_EXPORT)
+    set(_install_component "")
+  else()
+    set(_install_component COMPONENT ${_IBPF_COMPONENT})
+  endif()
```
Comment (drdanz) [inline modules/InstallBasicPackageFiles.cmake]: I don't understand why the `COMPONENT` should be influenced by `NO_EXPORT`, I think this and the following changes to the `COMPONENT` are not necessary...

File: modules/InstallBasicPackageFiles.cmake [hunk 2]
```diff
@@ -583,7 +597,7 @@ ${_compatibility_vars}
                                    ${_arch_independent})
   install(FILES "${_IBPF_EXPORT_DESTINATION}/${_version_filename}"
           DESTINATION ${_IBPF_INSTALL_DESTINATION}
-          COMPONENT ${_IBPF_COMPONENT})
+          ${_install_component})
```
Comment (drdanz) [inline modules/InstallBasicPackageFiles.cmake]: *

File: modules/InstallBasicPackageFiles.cmake [hunk 3]
```diff
@@ -719,18 +733,22 @@ endif()
   install(FILES "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/${_config_filename}.install"
           DESTINATION ${_IBPF_INSTALL_DESTINATION}
           RENAME ${_config_filename}
-          COMPONENT ${_IBPF_COMPONENT})
+          ${_install_component})
```
Comment (drdanz) [inline modules/InstallBasicPackageFiles.cmake]: *

File: modules/InstallBasicPackageFiles.cmake [hunk 4]
```diff
@@ -719,18 +733,22 @@ endif()
...
-          NAMESPACE ${_IBPF_NAMESPACE}
-          DESTINATION ${_IBPF_INSTALL_DESTINATION}
-          FILE "${_targets_filename}"
-          COMPONENT ${_IBPF_COMPONENT})
+  if(NOT _IBPF_NO_EXPORT)
+    install(${_install_cmd}
+            NAMESPACE ${_IBPF_NAMESPACE}
+            DESTINATION ${_IBPF_INSTALL_DESTINATION}
+            FILE "${_targets_filename}"
+            ${_install_component})
```
Comment (drdanz) [inline modules/InstallBasicPackageFiles.cmake]: *

File: modules/InstallBasicPackageFiles.cmake (hunk 1 above)
Comment (mbeutel) [reply modules/InstallBasicPackageFiles.cmake]: You are right, I just checked again, and this is not necessary. I had misunderstood an error message that was actually related to an `install()` command that needed to be suppressed.

Will fix.

**Comment by mbeutel** [comment]
I also added https://github.com/robotology/ycm/pull/268/commits/73d2ecb087d51ff3d8c6e16434ff9a94e0f48a76 which simplifies YCM's own installation by using `install_basic_package_files()` and `NO_EXPORT` as you suggested in #263.

File: CMakeLists.txt [hunk 5]
```diff
@@ -72,32 +72,14 @@ if(CMAKE_VERSION VERSION_LESS 3.14)
...
-                              INSTALL_DESTINATION "${CMAKE_INSTALL_PREFIX}/${YCM_INSTALL_CMAKE_DIR}"
-                              PATH_VARS YCM_MODULE_DIR
-                              NO_CHECK_REQUIRED_COMPONENTS_MACRO)
-
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/YCMConfigInstall.cmake"
-        DESTINATION "${YCM_INSTALL_CMAKE_DIR}"
-        RENAME YCMConfig.cmake)
+include(InstallBasicPackageFiles)
+install_basic_package_files(YCM
+                            VERSION ${YCM_VERSION}
```
Comment (drdanz) [inline CMakeLists.txt]: :heart_eyes:

**Comment by drdanz** [comment]
Merged, thanks! And thanks for adding the YCM installation part! :heart_eyes:

**Pull request merged**